packages/openapi-v3/test/unit/decorators/param/param-header.decorator.unit.ts

```diff
@@ -77,18 +65,163 @@ describe('Routing metadata for parameters', () => {
         @get('/greet')
         greet(@param.header.boolean('name') name: boolean) {}
       }
+      const expectedParamSpec = {
```

Comment:
it'd be cool if you can create a function which produces the expected parameter spec by feeding it the `in` and `type/format` values.
mock up:
```typescript
function getExpectedSchema(name?: string, in: string, type: string, format?: string): Object (a specific type) {
  const expectedParamSpec = {
    name: name || 'name',
    in: in,
    schema: {
      type: type
    },
  };
   if (format) expectedParamSpec.schema.format = format;
   return expectedParamSpec;
}
```

Reply:
@b-admike thank you for the code! Sometimes I don't extract the common code into a function for 2 reasons:
- For the simple expected result/object, I would rather see the result directly instead of a function that generates it when I read the code. The over complicated extraction makes my reading inconsistent :(
- And when the test fails, I can easily tell the unexpected part from the error msg, instead of jumping into a function to see what it does and what parameters it takes.

So personally I would prefer to keep those expected param specs as they are, they are not that nested and just around 5 lines :)

Reply:
Sounds good to me!